File: Backend/BackendTiendaMigrado/backendtienda/src/main/java/com/backend/backendtienda/dto/OrderDTOs.java
package com.backend.backendtienda.dto;

import java.math.BigDecimal;
import java.util.List;

public final class OrderDTOs {

    private OrderDTOs() {
    }

    // ===== requests =====
    public record CreateOrderRequest(
            BigDecimal totalAmount,
            List<CreateOrderDetailRequest> details) {
    }

    public record CreateOrderDetailRequest(
            Integer productId,
            Integer quantity,
            BigDecimal unitPrice) {
    }

    // ===== responses =====
    public record GetOrderResponse(
            Integer orderId,
            String date,                      // ← antes orderDate
            BigDecimal totalAmount,
            List<GetOrderDetailResponse> details) {
    }

    public record GetOrderDetailResponse(
            String imageUrl,
            String productName,
            Integer quantity,
            BigDecimal total) {               // ← antes subtotal
    }
}

File: Backend/BackendTienda/BackendTienda/DTOs/OrderDTOs.cs
namespace BackendTienda.DTOs
{
    public class OrderDTOs
    {
        #region requests
        public record CreateOrderRequest(
            decimal totalAmount,
            IEnumerable<CreateOrderDetailRequest> details
        );

        public record CreateOrderDetailRequest(
            int productId,
            int quantity,
            decimal unitPrice
        );
        #endregion

        #region responses

        public record GetOrderResponse(
            int orderId,
            string date,
            decimal totalAmount,
            IEnumerable<GetOrderDetailResponse> details
        );


        public record GetOrderDetailResponse(
            string imageUrl,
            string productName,
            int quantity,
            decimal total
        );
                #endregion
            }
        }
